srcCxx/shape_main.cxx

```diff
@@ -1029,7 +1032,7 @@ class ShapeApplication {
 #endif
 
                 if (retval == RETCODE_OK) {
-                    int i;
+                    unsigned int i;
                     for (i = 0; i < samples.length(); i++)  {
```

Comment:
If Fast DDS, `samples.length()` returns an `int32_t`. I suggest the following to let the compiler find the correct type:

```suggestion
                    auto n_samples = samples.length();
                    decltype(n_samples) i;
                    for (i = 0; i < n_samples; i++)  {
```

Reply:
The same happens with Connext. I agree this will work for all implementations

Reply:
Sounds good.  I'll change it -- I don't think we'll need to build this with an old compiler that doesn't have C++11 `auto`/`decltype`.